File: abc238/e/e.jl
function read_ints()::Vector{Int}
    line = readline()
    return map(x -> parse(Int, x), split(line))
end

n, q = read_ints()
adj = [Int[] for _ in 1:n+1]

for _ in 1:q
    u, v = read_ints()
    push!(adj[u], v + 1)
    push!(adj[v+1], u)
end

stk = Int[1]
vis = fill(false, n + 1)
vis[1] = true

while !isempty(stk)
    u = pop!(stk)
    for v in adj[u]
        if !vis[v]
            vis[v] = true
            push!(stk, v)
        end
    end
end

println(vis[n+1] ? "Yes" : "No")

File: abc238/e/e.cc
#include <cstdio>
#include <iostream>
#include <queue>
#include <vector>

using namespace std;

template <typename T> void read(T &x) {
  x = 0;
  char c = getchar();
  T sig = 1;
  for (; !isdigit(c); c = getchar())
    if (c == '-')
      sig = -1;
  for (; isdigit(c); c = getchar())
    x = (x << 3) + (x << 1) + c - '0';
  x *= sig;
}

class Solution {
public:
  void solve() {
    int n, q;
    read(n), read(q);
    vector<vector<int>> adj(n + 1);

    for (int i = 0; i < q; ++i) {
      int l, r;
      read(l), read(r);
      adj[l - 1].emplace_back(r);
      adj[r].emplace_back(l - 1);
    }

    vector<bool> vis(n + 1);
    vis[0] = true;
    queue<int> que;
    que.emplace(0);

    while (!que.empty()) {
      int u = que.front();
      que.pop();
      for (int v : adj[u]) {
        if (!vis[v]) {
          vis[v] = true;
          que.emplace(v);
        }
      }
    }

    if (vis[n]) {
      cout << "Yes" << endl;
    } else {
      cout << "No" << endl;
    }
  }
};

int main() {
  ios::sync_with_stdio(false);
  cin.tie(0);
  Solution solution = Solution();
  solution.solve();
}